### PR TITLE
Update `configparser` to 3.12

### PR DESCRIPTION
**Pull request by sobolevn**
Source commit https://github.com/python/cpython/commit/296e4efebbd1865153ed3be24887a2af3898a0c4

### CODE REVIEW

File: stdlib/configparser.pyi [hunk 1]
```diff
@@ -5,31 +5,53 @@ from re import Pattern
...
-    "ExtendedInterpolation",
-    "LegacyInterpolation",
-    "SectionProxy",
-    "ConverterMapping",
-    "DEFAULTSECT",
-    "MAX_INTERPOLATION_DEPTH",
-]
-
-if sys.version_info < (3, 12):
-    __all__ += ["SafeConfigParser"]
```
Comment (srittau) [inline stdlib/configparser.pyi:32]: What was wrong with this syntax? This looks far more readable to me.

Comment (sobolevn) [reply stdlib/configparser.pyi:32]: `__all__` was changed in 3.12 from `list` to `tuple`, I reflected that

Comment (AlexWaygood) [reply stdlib/configparser.pyi:32]: Yeah, unfortunately stubtest will (reasonably, if you squint at it the right way) complain if `__all__` is a tuple at runtime but a list in the stub

Comment (srittau) [reply stdlib/configparser.pyi:32]: Ah, I missed that.

**Review by srittau — APPROVED** [review]

**Pull request merged**